pandas/tests/groupby/test_groupby_iloc.py

```diff
@@ -0,0 +1,154 @@
+""" Test positional grouped indexing with iloc GH#42864"""
+
+import random
+
+import pandas as pd
+import pandas._testing as tm
+
+
+def test_doc_examples():
+    """Test the examples in the documentation"""
+
+    df = pd.DataFrame(
+        [["a", 1], ["a", 2], ["a", 3], ["b", 4], ["b", 5]], columns=["A", "B"]
+    )
+
+    grouped = df.groupby("A")
+    result = grouped.iloc[1:2, :]
+    expected = pd.DataFrame([["a", 2], ["b", 5]], columns=["A", "B"], index=[1, 4])
+
+    tm.assert_frame_equal(result, expected)
+
+    result = grouped.iloc[:-1, -1:]
+    expected = pd.DataFrame([1, 2, 4], columns=["B"], index=[0, 1, 3])
+
+    tm.assert_frame_equal(result, expected)
+
+
+def test_multiindex():
+    """Test the multiindex mentioned as the use-case in the documentation"""
+
+    def make_df_from_data(data):
```

Comment:
please use fixtures, the test should be as short as possible and easily understanable.

Reply:
Looks like I need to learn a bit more about pytest...
This test was more of a demo of the main use case. I can cut it down to a unit test (when I get back from vacation).